test/hotspot/jtreg/compiler/c2/irTests/TestIRStoreCorBAddIMask.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2022, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8282470
+ * @summary C2: Optimize patterns like s = (short) ((x << Imm) >> Imm + y) when Imm < 16 and apply the optimization to byte and char.
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.TestIRStoreCorBAddIMask
+ */
+public class TestIRStoreCorBAddIMask {
+
+    private static final Random RANDOM = Utils.getRandomInstance();
+
+    private static final int[] SPECIAL_IN = {
+        0, 0x1, 0x8, 0xF, 0x3F, 0x7C, 0x7F, 0x80, 0x81, 0x8F, 0xF3, 0xF8, 0xFF,
+        0x38FF, 0x3FFF, 0x8F8F, 0x8FFF, 0x7FF3, 0x7FFF, 0xFF33, 0xFFF8, 0xFFFF, 0xFFFFFF,
+        Integer.MAX_VALUE, Integer.MIN_VALUE
+    };
+
+    private byte BYTE_OUT = 0;
+    private char CHAR_OUT = 0;
+    private short SHORT_OUT = 0;
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @Arguments({Argument.DEFAULT, Argument.DEFAULT})
```

Comment:
NIT: Won't be better to use RANDOM_EACH in the parameters of this and other methods?